lib/widgets/context_menu/context_menu_entry.dart

```diff
@@ -0,0 +1,147 @@
+import 'package:files/widgets/context_menu/context_sub_menu_entry.dart';
+import 'package:flutter/material.dart';
+
+abstract class BaseContextMenuEntry extends PopupMenuEntry<String> {
+  /// Using for [represents] method.
+  final String id;
+
+  /// A widget to display before the title.
+  /// Typically a [Icon] widget.
+  final Widget? leading;
+
+  /// The primary content of the menu entry.
+  /// Typically a [Text] widget.
+  final Widget title;
+
+  /// bool flag to block gestures from user
+  final bool enabled;
+
+  const BaseContextMenuEntry({
+    required this.id,
+    required this.title,
+    this.leading,
+    this.enabled = true,
+    Key? key,
+  }) : super(key: key);
+
+  @override
+  double get height => 48;
```

Comment:
Mostly a nit but maybe find a way to make this vary based on the platform visual density? Would not want a mobile sized tile for desktop for example

Reply:
We can use `kMinInteractiveDimension` which used in `PopupMenuItem<T>` or
something like `_defaultTileHeight` for a `ListTile`. Both of it suggestions not really adaptive

Reply:
I think right now flutter itself has no good solution, either pass the height as a param (kinda yuck) or just leave as is for now